gtsfm/utils/metrics.py

```diff
@@ -0,0 +1,195 @@
+"""Utilities to compute and save evaluation metrics.
+
+Authors: Akshay Krishnan
+"""
+import json
+import numpy as np
+import os
+from dask.delayed import Delayed
+from typing import Dict, List, Optional, Tuple, Union
+
+from gtsam import Point3, Rot3, Pose3, Unit3
+
+import gtsfm.utils.geometry_comparisons as comp_utils
+from gtsfm.common.sfm_result import SfmResult
+
+# A StatsDict is a dict from string to optional floats or their lists.
+StatsDict = Dict[str, Union[Optional[float], List[Optional[float]]]]
+
+
+def get_errors_statistics(errors: List[Optional[float]]) -> StatsDict:
+    """Computes statistics (min, max, median) on the given list of errors
+
+    Args:
+        errors: List of errors for a metric.
+
+    Returns:
+        A dict with keys min_error, max_error, median_error,
+        and errors_list mapping to the respective stats.
+    """
+
+    metrics = {}
+    valid_errors = [error for error in errors if error is not None]
+    metrics["median_error"] = np.median(valid_errors)
+    metrics["min_error"] = np.min(valid_errors)
+    metrics["max_error"] = np.max(valid_errors)
+    metrics["errors_list"] = errors
+    return metrics
+
+
+def compute_rotation_angle_metrics(
```

Comment:
We have a similar function in `geometry_comparisons`. Can you check if we need an additional function.

Reply:
There is a function that compares two lists of rotations. 
There is another function that returns the angle between two individual rotations (which I call in this function). I could do all the looping within `compute_averaging_metrics`, but I think the code is more modular this way.

Reply:
Let me know if you have other ideas.

Reply:
i think it's ok